explorer/interpreter/value.h

```diff
@@ -773,15 +786,9 @@ class ConstraintType : public Value {
     Nonnull<const InterfaceType*> interface;
   };
 
-  using EqualityConstraint = Carbon::EqualityConstraint;
+  using RewriteConstraint = Carbon::RewriteConstraint;
```

Comment:
This seems a little odd to me, to have both defined outside the class then using with the same name inside. Maybe get rid of both this and EqualityConstraint?

Reply:
If we do that, I'd want to move the other two structs out of the class too, for consistency. Will look into that in a separate PR.